homeassistant/components/shopping_list/__init__.py

```diff
@@ -19,7 +19,8 @@
 DOMAIN = "shopping_list"
 _LOGGER = logging.getLogger(__name__)
 CONFIG_SCHEMA = vol.Schema({DOMAIN: {}}, extra=vol.ALLOW_EXTRA)
-EVENT = "shopping_list_updated"
+EVENT_UPDATE = "shopping_list_updated"
+EVENT_ITEM_ADD = "shopping_list_item_added"
```

Comment:
We should not make this change. Instead add an `action` attribute to the event data of the existing event. Also don't forget to add tests.

Reply:
Updated as you suggested. Added an "action" and an "item" attribute to the shopping_list_updated event. Tested in my environment and it works as expected. Attempting TOX tests now! Thanks @balloob!